### PR TITLE
Migrate membership, nicks, scored-pool and session to decl_error

### PR DESCRIPTION
**Pull request by DarkEld3r**
Part of the #4365 issue.

### CODE REVIEW

File: frame/scored-pool/src/lib.rs [hunk 1]
```diff
@@ -222,11 +222,27 @@ decl_event!(
 	}
 );
 
+decl_error! {
+	/// Error for the scored-pool module.
+	pub enum Error for Module<T: Trait<I>, I: Instance> {
+		/// Already a member.
+		AlreadyInPool,
+		/// Balance is too low to submit candidacy.
+		InsufficientBalance,
```
Comment (shawntabrizi) [inline frame/scored-pool/src/lib.rs]: Do we need this error?

Balances module already has one

File: frame/scored-pool/src/tests.rs [hunk 2]
```diff
@@ -223,9 +220,9 @@ fn index_mismatches_should_abort() {
 	new_test_ext().execute_with(|| {
 		let who = 40;
 		let index = 3;
-		assert_noop!(ScoredPool::withdraw_candidacy(Origin::signed(who), index), INDEX_ERR);
-		assert_noop!(ScoredPool::score(Origin::signed(ScoreOrigin::get()), who, index, 99), INDEX_ERR);
-		assert_noop!(ScoredPool::kick(Origin::signed(KickOrigin::get()), who, index), INDEX_ERR);
+		assert_noop!(ScoredPool::withdraw_candidacy(Origin::signed(who), index), Error::<Test, DefaultInstance>::WrongAccountIndex);
+		assert_noop!(ScoredPool::score(Origin::signed(ScoreOrigin::get()), who, index, 99), Error::<Test, DefaultInstance>::WrongAccountIndex);
```
Comment (shawntabrizi) [inline frame/scored-pool/src/tests.rs]:
```suggestion
		assert_noop!(ScoredPool::score(Origin::signed(ScoreOrigin::get()), who, index, 99), Error::<Test, _>::WrongAccountIndex);
```

Can skip writing out `DefaultInstance` I believe

File: frame/scored-pool/src/lib.rs (hunk 1 above)
Comment (DarkEld3r) [reply frame/scored-pool/src/lib.rs]: Nice catch, though `InsufficientBalance` from balances is not in the master yet. :wink: 
The test in `scored-pool` should be updated too, so I'm going to fix this after your pull request is merged.

**Comment by DarkEld3r** [comment]
Fixed.

**Pull request merged**